Src/FluentAssertions/Collections/SelfReferencingCollectionAssertions.cs

```diff
@@ -646,8 +647,9 @@ public AndConstraint<TAssertions> NotContain(Expression<Func<T, bool>> predicate
         {
             Guard.ThrowIfArgumentIsNull(predicate, nameof(predicate));
 
+            var body = Regex.Replace(predicate.Body.ToString(), @"{(\d+)}", @"{{$1}}");
```

Comment:
We have a `EscapePlaceholders()` extension method on `string`, which seems applicable here to get the same effect.

Reply:
Thanks Jonas, I replaced the call. Just one a smale note - that method doesn't check the number inside the brackets so that we can affect some valid symbols (anyway it shouldn't be a problem).